contributing/development/handling_compatibility_breakages.rst

```diff
@@ -0,0 +1,158 @@
+.. _doc_handling_compatibility_breakages:
+
+Handling compatibility breakages
+=================================
+
+So you've added a new parameter to a method, changed the return type,
+changed the type of a parameter, or changed its default value,
+and now the automated testing is complaining about compatibility breakages?
+
+Not to worry, there's a system to handle just that!
+
+Compatibility is handled in a few different ways. In GDScript this is simple,
+new arguments are usually given default values when added, so existing scripts
+won't break in such cases.
+
+However in other cases, like GDExtension and C#, the exact way the method is
+declared matters and must be matched exactly. This is handled by creating
+compatibility methods that then get mapped to the new versions of these
+methods. All of this happens *almost* automatically, there are just a few
+things you need to do.
+
+Binding compatibility methods
+-----------------------------
+
+Just like how you use ``ClassDB`` to bind methods, there's a way to bind
+compatibility methods, and conveniently these are named very similarly to their
+ordinary counterparts.
+
+.. code-block:: cpp
+
+    ClassDB::bind_method(D_METHOD("my_method", "arg1", "arg2"), &MyClass::my_method);
+    ClassDB::bind_compatibility_method(D_METHOD("my_method", "arg1", "arg2", "arg3"), &MyClass::_my_method_bind_compat_1234, DEFVAL(0));
```

Comment:
While this is a totally valid example of an argument being removed, I feel like the reverse case is more common: adding a new argument to method. So, I'd personally reverse these:

```suggestion
    ClassDB::bind_method(D_METHOD("my_method", "arg1", "arg2", "arg3"), &MyClass::my_method, DEFVAL(0));
    ClassDB::bind_compatibility_method(D_METHOD("my_method", "arg1", "arg2"), &MyClass::_my_method_bind_compat_1234);
```